4-auth/tests/test_auth.py

```diff
@@ -120,3 +122,13 @@ def test_mine(self, model, client_with_credentials):
         body = rv.data.decode('utf-8')
         assert 'Book 1' in body
         assert 'Book 2' not in body
+
+    @mock.patch("httplib2.Http")
+    def test_oauth2_callback(self, HttpMock):
```

Comment:
This isn't really testing `oauth2_callback`, it's testing `_request_user_info`. In general, I'd like it to test the latter by calling the former, but it's up to you.